src/granite_orm/transactions.cr

```diff
@@ -7,9 +7,21 @@ module Granite::ORM::Transactions
     @updated_at : Time?
     @created_at : Time?
 
+    # The import class method will run a batch INSERT statement for each model in the array
+    # the array must contain only one model class
+    # invalid model records will be skipped
+    def self.import(model_array)
+      raise ArgumentError.new("Model class mismatch:  expected array of only #{self} models.") unless model_array.all? { |model| model.class == self }
```

Comment:
Can this definition be changed so that the compiler will enforce the type constraint instead of a runtime error?

First guess, something like this might work:

```crystal
def self.import(model_array : Array(self))
```

Reply:
Ooo yes!  I will do that tonight (still getting used to static typed land :P)